### PR TITLE
add some types to os

### PR DESCRIPTION
**Pull request by tharvik**
Add some missing type annotation to os.

I choosed to remove `__init__` from `stat_result` because it isn't documented and it isn't design to be used by an user anyway.

### CODE REVIEW

**Comment by matthiaskramm** [comment]
Grepping through our codebase, I found a lot of occurrences of manual `stat_result` construction. Sure, it's add odd thing to do, but people do it anyway, so type checkers shouldn't whine about it.

**Comment by tharvik** [comment]
>Grepping through our codebase, I found a lot of occurrences of manual stat_result construction. Sure, it's add odd thing to do, but people do it anyway, so type checkers shouldn't whine about it.

Updated, from what I saw, it takes a tuple of 10 int.

File: stdlib/3/os/__init__.pyi [hunk 1]
```diff
@@ -124,7 +124,10 @@ class stat_result:
     st_ctime = 0.0 # platform dependent (time of most recent metadata change
                    # on  Unix, or the time of creation on Windows)
 
-    def __init__(self, tuple) -> None: ...
+    # not documented
+    def __init__(self,
+                 tuple: Tuple[int, int, int, int, int, int, int, int, int, int]) \
```
Comment (matthiaskramm) [inline stdlib/3/os/__init__.pyi]: According to posixmodule.c, this takes 10-16 integers, depending on what kind of system you're on.

I'd suggest `Tuple[int, ...]`.

Comment (tharvik) [inline stdlib/3/os/__init__.pyi]: Weird to have variable length for a struct like this; anyway, `Tuple[int, ...]` it is.

**Pull request merged**